model.js

```diff
@@ -21,3 +21,29 @@ exports.selectArticle = (id) => {
       }
     });
 };
+
+exports.selectAllArticles = () => {
+  return db
+    .query(
+      "SELECT author, title, article_id, topic, created_at, votes, article_img_url FROM articles ORDER BY created_at DESC"
+    )
+    .then((allArticles) => {
+      console.log(allArticles.rows, "<--- here");
+      return allArticles.rows;
+    });
+};
+
+exports.addComments = (articlesArray) => {
+  const promises = articlesArray.map((article) => {
+    const articleID = article.article_id;
+
+    return db
+      .query("SELECT * FROM comments WHERE article_id = $1", [articleID])
+      .then((articleComments) => {
+        article.comment_count = articleComments.rows.length;
+      });
+  });
+  return Promise.all(promises).then(() => {
+    return articlesArray;
+  });
```

Comment:
There's quite a lot going on in here. Have a think about how you could refactor these models to get the comments at the same time as the articles in one sql query. Have a look at these notes as they may help you https://l2c.northcoders.com/courses/sd-notes/back-end#sectionId=aggregate-functions,step=intro